### ISSUE TITLE
Roadmap or Project Direction?

### ISSUE THREAD

**Issue by pha3z** [canton7/Stylet#62]
I've been extremely impressed by the Reddit Browser sample.  That is a very good demo application that covers a good breadth of features and shows a practical real world use case!  That sample alone teaches so much about the framework.

After having looked at MVVM Cross and seeing all the pain that would be incurred by trying to use Xamarin  as a foundation, I've decided it would make more sense to use Stylet on our initial WPF version of our application.  If and when we decide to do target versions for other OSes, we'll consider Xamarin for those cases.

I'm thoroughly impressed with the documentation and samples and the way in which the framework operates. It made me wonder about the size of Stylet's userbase. It seems to be relatively off-the-radar compared to other MVVMs (which is ironic IMO because I think its superior to the ones I've seen). I wondered what the author's opinion is on that.  Do you want it to stay used by a relatively small group of users strictly as an open source option? Or have you considered doing commercial support for it or something that would grow its presence and help ensure its maintenance?

Personally, I'd like to see the author somehow profit from the framework, if possible, because I believe its that good and monetary compensation certainly helps ensure continued maintenance/development.  I do realize that not ever project is about making money though. Having a simple and effective MVVM strictly as an open source initiative makes sense to me, assuming the author feels the reward of overseeing its maintenance is worth it.

Thoughts?

**Comment by canton7** [comment]
Yeah, it's relatively small. I wrote it because I liked Caliburn.Micro, but didn't like the amount of magic, plus it was good challenge. I've used it in a few products at my day job, and it gets a level of maintenance. However I'm not too interested in growing it, or trying to actively promote it: I like the fact that the feature set is small and minimal (that was one of the original aims), and I simply don't have the resources to try and out-promote the other (all very good) MVVM frameworks out there.

Plus, promoting something too widely necessarily means that you end up spending your time helping people who are beginners at C#, or who have never used MVVM before!

Trying to do commercial support for an open-source product sounds like my worst nightmare: I've got a great job making cool stuff which provides a solid and steady income, and I don't see the point in dedicating my free time acting as a helpdesk on top of that.

If I was going to try and monetise open source work, I've got other projects which are far more popular. I did add donations to one project (with 3 orders of magnitude more users that Stylet), but the amount that comes in is nowhere near enough to even consider moving off the day job as a remote possibility.

So, I think Stylet is and will remain to be one of the many niche little MVVM frameworks out there. People who use it tend to love it, but people who use the other frameworks also get along just fine as well.

**Comment by pha3z** [comment]
That makes a lot of sense. Thanks for sharing your perspective.  :)

**Comment by canton7** [comment]
To answer the "Roadmap or Project Direction?" bit: because I want to keep the scope of Stylet small (that's one its USPs) there's no planned future development. I'll respond to help requests, fix things that are wrong, and make small additions, but there's no active development towards some larger goal.

**Comment by pha3z** [comment]
That makes sense.  And sounds good.  I agree about the USP: the small scope of Stylet is one of the things that attracted me to it.  It's very easy to get started with and looks like it will work very well for a small-ish to medium scale application.

**Comment by wakuflair** [comment]
Though it's nothing to do with this issue, as a  Stylet user I want to say thanks to @canton7 .
Thank you for your effort with Stylet! It helps me a lot!
